packages/database/prisma/schema.prisma

```diff
@@ -154,6 +155,14 @@ model MerchantProduct {
   merchantPrices    MerchantPrice[]
 }
 
+model MerchantPurchase {
+  id                String          @id @default(uuid())
+  checkoutSessionId String
+  merchantAccountId String
+  merchantAccount   MerchantAccount @relation(fields: [merchantAccountId], references: [id], onDelete: Cascade)
+  purchase          Purchase[]
+}
```

Comment:
a new table for storing the `checkoutSessionId` related to a `Purchase`.

Reply:
we've gone to pretty great lengths to not encode stripe specific data in the context of "merchant" in this schema

IMO probably need a table for this and not a simple field. 

MerchantSession or similar would fit the pattern with an "identifier"

Hard to answer the "what about paypal" question, but "checkoutSessionId" is definitely very stripe

Reply:
that makes a lot of sense. looking more closely at the comment you made on an earlier PR, I see you had suggested `identifier` there as well. https://github.com/skillrecordings/products/pull/662#discussion_r1043683711

![CleanShot 2023-02-15 at 12 10 40@2x](https://user-images.githubusercontent.com/694063/219116765-e1c812e6-abd3-46e4-9f8e-b816cbd6d8ea.png)

So I guess how this could work then is: if I have a `Purchase` that is tied to a `MerchantSession` whose `MerchantAccount` is `stripe`, then the `MerchantSession.identifier` is expected to be a `checkoutSessionId` and I can use it as such.